src/lib/services/transfer-service/TransferService.ts

```diff
@@ -169,6 +204,80 @@ export class TransferService {
         }
     }
 
+    private async putThumbnailImageToShell(aas: AssetAdministrationShell, apikey?: string): Promise<TransferResult> {
+        try {
+            const aasThumbnail = await this.sourceAasRepositoryClient.getThumbnailFromShell(aas.id);
+            const fileName = ['thumbnail', generateRandomId()].join('');
+            await this.targetAasRepositoryClient.putThumbnailToShell(aas.id, aasThumbnail, fileName, {
+                headers: {
+                    Apikey: apikey,
+                },
+            });
+            return { success: true, operationKind: 'AasRepository', resourceId: 'Thumbnail transfer.', error: '' };
+        } catch (e) {
+            return {
+                success: false,
+                operationKind: 'AasRepository',
+                resourceId: 'Thumbnail transfer.',
+                error: e.toString(),
+            };
+        }
+    }
+
+    private async processAttachments(submodelId: string, attachmentDetails: AttachmentDetails[], apikey?: string) {
+        const promises = [];
+
+        for (const attachmentDetail of attachmentDetails) {
+            try {
+                attachmentDetail.file = await this.sourceSubmodelRepositoryClient.getAttachmentFromSubmodelElement(
+                    submodelId,
+                    attachmentDetail.idShortPath,
+                );
+                this.processFileExtension(attachmentDetail);
```

Comment:
I dont like function that perform side effects to inputs. I'd rather have the assignment explicit, e.g., attachmentDetail.fileName=extractFileExtensionFromFileType(...)

Reply:
We will change this in the API merge